### DISCUSSION TITLE
Access SQLite views EOI

### DISCUSSION THREAD

**Discussion by conradbez** [pocketbase/pocketbase#300] [Q&A] [answered]
Awesome project! I’ve been using this in conjunction with a dbt (sql modeling) backend but I’m running into the issue of having to turn my sql logic into tables to work with pocketbase. Ideally, for my use case, pocketbase would treat views as read only. 

Interested if access to SQLite views would be a valid feature for me to investigate or if this is outside the intended scope for this project.

**Comment by ganigeorgiev** [comment, marked as answer]
There are plans to add support for virtual/computed/rawsql fields that could be used to create a read-only collections (primarily to allow simple reporting/aggregations - https://github.com/pocketbase/pocketbase/discussions/265), but for now I don't plan adding support for read-only views since I'm not sure whether the added complexity (both in the code and the UI) will be worthed.

In order to make your plain SQL table to appear and allow PocketBase to manage them, you must create a Collection for each table. In the future I may consider implementing "autodiscovery" cli command that will try to create a default collection from plain sql table, but a lot of other tasks already have piled in the [roadmap](https://github.com/orgs/pocketbase/projects/2), so for now this is a very low priority.
